src/lang/modifyAst/boolean.ts

```diff
@@ -75,33 +76,12 @@ export async function applyUnionFromTargetOperatorSelections(
   }
 ): Promise<Error | void> {
   const ast = dependencies.kclManager.ast
-
-  const artifacts: Artifact[] = []
-  for (const selection of solids.graphSelections) {
-    if (selection.artifact) {
-      artifacts.push(selection.artifact)
-    }
-  }
-
-  if (artifacts.length < 2) {
-    return new Error('Not enough artifacts selected')
-  }
-
-  const orderedChildrenEach = artifacts.map((artifact) =>
-    findAllChildrenAndOrderByPlaceInCode(
-      artifact,
-      dependencies.kclManager.artifactGraph
-    )
+  const lastVars = getLastVariableDeclarationsFromSelections(
+    solids,
+    ast,
+    dependencies.kclManager.artifactGraph
   )
-
-  const lastVars: VariableDeclaration[] = []
-  for (const orderedArtifactLeaves of orderedChildrenEach) {
-    const lastVar = getLastVariable(orderedArtifactLeaves, ast)
-    if (!lastVar) continue
-    lastVars.push(lastVar.variableDeclaration.node)
-  }
```

Comment:
This was repeated across the other two and now three with this PR so I figured it'd be nicer in a little util fn